R/utils.R

```diff
@@ -13,6 +13,48 @@ setup_project <- function() {
   source(path)
 }
 
-use_r_packages <- function(path = "deduplicate/load-and-attach-r-packages.R") {
+use_r_packages <- function(path = packages_path()) {
   source(path)
+  conflicted::conflict_prefer("filter", "dplyr")
+  conflicted::conflict_prefer("lag", "dplyr")
+
+  invisible(path)
+}
+
+packages_path <- function() {
+  "deduplicate/load-and-attach-r-packages.R"
+}
+
+create_install_calls_for_dockerfile <- function(path = packages_path()) {
+  pkg <- sub("library\\((.*)\\)", "\\1", readLines(path, encoding = "UTF-8"))
+  glue('    && Rscript -e "install.packages(\'{pkg}\')" \\')
+}
+
+update_dockerfile_packages <- function(path = "docker/2diirunner-with-packages/Dockerfile") {
+  dkr <- readLines(path, encoding = "UTF-8")
+
+  raw <- readLines(packages_path(), encoding = "UTF-8")
+  pkg <- sub("library\\((.*)\\)", "\\1", raw)
+  pkg <- glue('    && Rscript -e "install.packages(\'{pkg}\')" \\')
+
+  out <- c(
+    dkr[1:marker_start(dkr)],
+    pkg,
+    dkr[marker_end(dkr):length(dkr)]
+  )
+  writeLines(out, path)
+
+  invisible(path)
+}
+
+marker_start <- function(lines) {
+  grep("# update-dockerfile-packages-start", lines)
+}
+
+marker_end <- function(lines) {
+  grep("# update-dockerfile-packages-end", lines)
+}
+
```

Comment:
I don't understand what these are for

Reply:
These are markers that the function `update_dockerfile_packages()` looks for in the Dockerfile -- so it knows what to update.
Thanks for noting it's unclear.